src/cargo/util/toml/targets.rs

```diff
@@ -526,7 +541,18 @@ fn to_bench_targets(
     Ok(result)
 }
 
-fn is_normalized(toml_targets: Option<&Vec<TomlTarget>>, autodiscover: Option<bool>) -> bool {
+fn is_normalized(toml_target: Option<&TomlTarget>, autodiscover: Option<bool>) -> bool {
+    if autodiscover != Some(false) {
+        return false;
+    }
+
+    let Some(toml_target) = toml_target else {
+        return true;
+    };
+    toml_target.name.is_some() && toml_target.path.is_some()
```

Comment:
What about this?

(Guess we'll need to turn `Option<&Vec<TomlTarget>` to `Option<&[TomlTarget]>` in are_normalized` if going to do that)

```suggestion
    are_normalized(toml_target.map(std::slice::from_ref), autodiscover)                                                                     
```